src/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.cpp

```diff
@@ -308,6 +320,48 @@ Propagator Propagator::FromEnvironment(
     };
 }
 
+void Propagator::registerDynamicsInformation(const Shared<Dynamics>& aDynamics)
+{
+    // Store read coordinate subsets information
+    Array<Pair<Index, Size>> readInfo = Array<Pair<Index, Size>>::Empty();
+
+    Size reducedStateSize = 0;
+    for (const Shared<const CoordinatesSubset>& subset : aDynamics->getReadCoordinatesSubsets())
+    {
+        Pair<Index, Size> indexAndSize = {this->coordinatesBrokerSPtr_->addSubset(subset), subset->getSize()};
+        readInfo.add(indexAndSize);
+
+        reducedStateSize += indexAndSize.second;
+    }
+
+    // Store write coordinate subsets information
+    Array<Pair<Index, Size>> writeInfo = Array<Pair<Index, Size>>::Empty();
+    for (const Shared<const CoordinatesSubset>& subset : aDynamics->getWriteCoordinatesSubsets())
+    {
+        Pair<Index, Size> indexAndSize = {this->coordinatesBrokerSPtr_->addSubset(subset), subset->getSize()};
+        writeInfo.add(indexAndSize);
+    }
+
+    this->dynamicsInformation_.add({aDynamics, readInfo, writeInfo, reducedStateSize});
+}
+
+NumericalSolver::StateVector Propagator::extractCoordinatesFromStateVector(const State& aState) const
+{
+    const State referenceFrameTransformedState = aState.inFrame(INTEGRATION_REFERENCE_FRAME);
```

Comment:
Might be worth adding a test where:

1. Initial state is in frameA (other than GCRF)
2. It's passed on to the propagator
3. It's propagated
4. We check that it makes sense

Essentially testing that we can give the propagator a State in any frame we desire.